packages/core-data/src/actions.js

```diff
@@ -139,6 +144,52 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Action triggered to delete an entity record.
+ *
+ * @param {string} kind              Kind of the deleted entity.
+ * @param {string} name              Name of the deleted entity.
+ * @param {string} recordId          Record ID of the deleted entity.
+ * @param {?Object}query             Special query parameters for the DELETE API call.
+ */
+export function* deleteEntityRecord( kind, name, recordId, query ) {
+	const entities = yield getKindEntities( kind );
+	const entity = find( entities, { kind, name } );
+	if ( ! entity ) {
+		return;
+	}
+
+	try {
+		let path = `${ entity.baseURL }/${ recordId }`;
+
+		if ( query ) {
+			path = addQueryArgs( path, query );
+		}
+
+		yield apiFetch( {
+			path,
+			method: 'DELETE',
+		} );
+
+		yield removeItems( kind, name, recordId, true );
+	} catch ( _error ) {
+		const persistedRecord = yield select(
+			'getEditedEntityRecord',
+			kind,
+			name,
+			recordId
+		);
+
+		yield receiveEntityRecords(
+			kind,
+			name,
+			persistedRecord,
+			undefined,
+			false
+		);
```

Comment:
We don't need to do this since we are not optimistically removing items before an API response.